### PR TITLE
fix(Scripts/AlteracValley): Mini bosses should evade with boss.

### LINKED ISSUE #10460: [Alterac Valley] You can pull mini bosses at last boss room without him (reset)
https://github.com/chromiecraft/chromiecraft/issues/2897

### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

It is possible to pull 1 by 1 of boss adds and reseting boss/adds but only to keep add you're killing.
That way, people destroy 2 towers, boss is then guarded by 2 guards, tank pull 1, they reset boss and another guard, and kill guard solo.

### Expected Blizzlike Behaviour

Whenever boss resets, adds should as well.

### Source

_No response_

### Steps to reproduce the problem

1. Need more people for this, can't reproduce it solo on local AC prob
2. Pull one of adds
3. Get out of range carefully, use spells like vanish, FD when you have aggro on boss
4. Boss resets and his adds continues to get killed by group

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/f3774fd9ac8241994396426e6c5f9246ce56400b

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-save-inventory](https://github.com/Nyeriah/mod-save-inventory)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #10460

## Issues Addressed:

- Closes #10460
- Closes https://github.com/chromiecraft/chromiecraft/issues/2897

## Tests Performed:

- Not tested.

## How to Test the Changes:

Need more people for this, can't reproduce it solo on local AC prob
Pull one of adds
Get out of range carefully, use spells like vanish, FD when you have aggro on boss
Boss resets and his adds reset as well.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by temperrr** [comment]
Tested:

I can still pull with hunter, taunt boss with tank, drop aggro on hunter , adds go back, boss stays.

**Comment by UltraNix** [comment]
Closes as it's not a bug.

**Comment by temperrr** [comment]
Reopening as it is a bug.

**Comment by temperrr** [comment]
Tested:

image


how: only target the main boss, the rest is not tied to it.

**Comment by UltraNix** [comment]
But you are testing something that is not related to the reported issue and it's not covered by this PR.

**Comment by UltraNix** [comment]
@temperrr Try again please.

**Comment by temperrr** [comment]
Fair enough, I was under the impression it would fix both. My bad on that. I'll test tomorrow again!

**Comment by temperrr** [comment]
When pulling server crashed with this in log:
zsh: segmentation fault (core dumped)  ./worldserver

**Comment by UltraNix** [comment]
@temperrr Do you have crashlog?

**Comment by temperrr** [comment]
That was all that was in it

**Comment by UltraNix** [comment]
Is your core built in `RelWitDebInfo` mode?

**Comment by temperrr** [comment]
yes trying again

**Comment by temperrr** [comment]
zsh: segmentation fault (core dumped)  ./worldserver

**Comment by temperrr** [comment]
image

**Comment by UltraNix** [comment]
@temperrr You are trying to kill/aggro/evade alliance or horde boss/mini bosses?

**Comment by temperrr** [comment]
Alliance one, I shoot afireball at m, then I get the aggro lines and crash

**Comment by UltraNix** [comment]
@temperrr Try now.

**Comment by temperrr** [comment]
Same thing :/

**Comment by UltraNix** [comment]
@temperrr Just attacked alliance mini boss - no crash.

**Comment by temperrr** [comment]
Ok then I am happy

**Comment by UltraNix** [comment]
@temperrr So it's okay now?

**Comment by acidmanifesto** [comment]
image

File: src/server/scripts/EasternKingdoms/AlteracValley/boss_vanndar.cpp [hunk 1]
```diff
@@ -59,11 +60,57 @@ class boss_vanndar : public CreatureScript
             YellTimer = urand(20 * IN_MILLISECONDS, 30 * IN_MILLISECONDS);
         }
 
-        void EnterCombat(Unit* /*who*/) override
+        void EnterCombat(Unit* victim) override
         {
             Talk(YELL_AGGRO);
         }
 
+        void AttackStart(Unit* victim) override
```
Comment (acidmanifesto) [inline src/server/scripts/EasternKingdoms/AlteracValley/boss_vanndar.cpp:68]: image

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

**Review by acidmanifesto — APPROVED** [review]

**Pull request merged**